### PR TITLE
Convert tick-setting methods to docstrings

### PR DESCRIPTION
**Pull request by dstansby**

### CODE REVIEW

File: lib/matplotlib/axes/_base.py [hunk 1]
```diff
@@ -2993,24 +2993,39 @@ def set_xticks(self, ticks, minor=False):
...
         Set the x ticks with list of *ticks*
 
-        ACCEPTS: sequence of floats
+        Parameters
+        ----------
+        ticks : sequence
+            Sequence of ticks
+
+        minor : bool, optional
+            If ``False`` sets major ticks, if ``True`` sets minor ticks.
```
Comment (tacaswell) [inline lib/matplotlib/axes/_base.py]: I think this can also take the string `'both'`.

File: lib/matplotlib/axes/_base.py [hunk 2]
```diff
@@ -2993,24 +2993,39 @@ def set_xticks(self, ticks, minor=False):
...
     def get_xmajorticklabels(self):
         """
-        Get the xtick labels as a list of :class:`~matplotlib.text.Text`
-        instances.
+        Get the xtick major labels
+
+        Returns
+        -------
+        labels : list
+            List of :class:`~matplotlib.text.Text` instancess
```
Comment (anntzer) [inline lib/matplotlib/axes/_base.py]: "instacesssss"

File: lib/matplotlib/axes/_base.py [hunk 3]
```diff
@@ -2993,24 +2993,39 @@ def set_xticks(self, ticks, minor=False):
         """
         Set the x ticks with list of *ticks*
 
-        ACCEPTS: sequence of floats
+        Parameters
+        ----------
+        ticks : sequence
+            Sequence of ticks
```
Comment (anntzer) [inline lib/matplotlib/axes/_base.py]: I would just say list... and list of floats (I guess?), as it's not a list of Tick objects (unless I got this completely wrong).

File: lib/matplotlib/axes/_base.py (hunk 1 above)
Comment (tacaswell) [reply lib/matplotlib/axes/_base.py]: on second thought, that makes not much sense for this `set_` method.

File: lib/matplotlib/axes/_base.py (hunk 3 above)
Comment (dstansby) [reply lib/matplotlib/axes/_base.py]: Could be a list of datetimes if it's a date axis?

Comment (anntzer) [reply lib/matplotlib/axes/_base.py]: Possibly, no idea.
But "ticks" without further qualification definitely sounds like "Tick objects" to me.

**Comment by dstansby** [comment]
Backported to `2.1.x` via. https://github.com/matplotlib/matplotlib/commit/172daf85529ed466a367dfd3a5e07ebe66da8c20

**Pull request merged**